### PR TITLE
Refactor: refactored createFullfacetPipeline function

### PR DESCRIPTION
**Pull request by Junjiequan**
## Description

Improved readability by reducing use case of else if condition

## Motivation

It was extremely hard to read what's going on inside createFullfacetPipeline function

## Tests included/Docs Updated?

- [ ] Included for each change/fix?
- [ ] Passing? (Merge will not be approved unless this is checked) 
- [ ] Docs updated?
- [ ] New packages used/requires npm install? 
- [ ] Toggle added for new features?

### CODE REVIEW

File: src/common/utils.ts [hunk 1]
```diff
@@ -633,11 +691,186 @@ export const createFullfacetPipeline = <T, Y extends object>(
...
-  pipeline.push({ $facet: facetObject });
+  pipeline.push({
+    $facet: facetObject as Record<string, PipelineStage.FacetPipelineStage[]>,
+  });
 
-  return pipeline as PipelineStage[];
+  return pipeline;
 };
 
+// NOTE: remove below comments in the future once its sure that everything is
```
Comment (bpedersen2) [inline src/common/utils.ts]: I would drop this ( git  handles restoration for us ;) )

**Review by bpedersen2 — CHANGES_REQUESTED** [review]
Check the eslint comment( T is unsed in handleScientifcQuery) on line 564

**Review by nitrosx — APPROVED** [review]

**Comment by nitrosx** [comment]
@Junjiequan Can you please resolve the conflict?

File: src/common/utils.ts (hunk 1 above)
Comment (nitrosx) [reply src/common/utils.ts]: Resolved!!!

**Pull request merged**